distribution/lib/Standard/Base/0.0.0-dev/src/Data/Read/Many_Files_List.enso

```diff
@@ -19,4 +21,4 @@ type Many_Files_List
 
 ## PRIVATE
 Many_Files_List.from (that : Vector) =
-    Many_Files_List.Value that that
+    Many_Files_List.Value that that Return_As_Base.As_Vector
```

Comment:
I think the default for files should be As_Merged_Table

Reply:
We cannot do that, because `As_Merged_Table` is only available in `Table` library. This has to work with just the `Base`.

_Technically_, we could make it so that _once `Table` is imported_, we change from `As_Vector` to `As_Merged_Table`. But I **really** don't like the concept that adding an import makes our code change behaviour in such ways...

(Well we do change behaviour as e.g. adding a `Table` import will make `Auto_Detect` able to parse CSV, whereas without that import it would have failed - but IMO change from 'fails' to 'succeeds' is ok, but from one return type to another is problematic. Open to discussion though.)

Reply:
With my product manager hat on. This has to be the default for when I am reading in multiple Excel workbooks. The other options just don't make any sense in the GUI.

With my engineering hat on, I'm not entirely sure how we solve this.

Reply:
I see.

1. What is your opinion on changing the default behaviour based on the imports? 
Engineering-wise that can surely be implemented quite easily, I was just hesitant as I don't like such behaviour change... But I don't have many better ideas. In GUI the `Table` will be imported 99% of the time, so I guess the behaviour will be fairly consistent, we can document it.

2. _Alternatively_, we could make `return` a required argument (throw the `Missing_Argument` error) and then the user would choose explicitly every time. We can even make `As_Merged_Table` the 'top' result in dropdown so it is easily accessible. Although I feel like having to set this argument every time instead of having a default will be quite annoying...

3. Third option is to keep the behaviour change less disruptive. As noted above - I think behaviour change that goes from 'error' to 'working' when an import is added (e.g. like with new supported formats in `Auto_Detect` made available by imports) is acceptable - it is much less disruptive than the type of result in unexpected ways. So we could make it so that the default is `As_Merged_Table` but if `Table` is not imported it raises an error:
    > Table output was selected, but the Table library is not imported. Please either import `Standard.Table` or change the return type to `As_Vector`.

-------

I feel like (2) is nicely consistent but too annoying in our most usual case.

I like (3) the most. I feel like it is the best compromise we can get. In the most popular use case (GUI with Table basically always imported) it works correctly and does not require unnecessary setting of arguments. In rare cases in GUI where the import is missing, it gives a clear error that tells the user possible resolutions - either add the import or explicitly request `Vector` return type.

Reply:
If we made the default `..As_Merged_Table` code, the resolution would be late and so would work if the Table is loaded into scope.

We can add a special case in the resolve handler so if not resolved it tries to resolve something which would let us throw and error saying missing the table input.

Reply:
After discussion it seems that (3) is what we want